daemon/cmd/endpoint.go

```diff
@@ -214,17 +222,17 @@ func (d *Daemon) createEndpoint(ctx context.Context, epTemplate *models.Endpoint
 
 	ep, err := endpoint.NewEndpointFromChangeModel(d.ctx, d, d.l7Proxy, d.identityAllocator, epTemplate)
 	if err != nil {
-		return invalidDataError(ep, fmt.Errorf("unable to parse endpoint parameters: %s", err))
+		return d.invalidDataError(ep, false, fmt.Errorf("unable to parse endpoint parameters: %s", err))
 	}
 
 	oldEp := d.endpointManager.LookupCiliumID(ep.ID)
 	if oldEp != nil {
-		return invalidDataError(ep, fmt.Errorf("endpoint ID %d already exists", ep.ID))
+		return d.invalidDataError(ep, true, fmt.Errorf("endpoint ID %d already exists", ep.ID))
```

Comment:
Good catch, will fix

Reply:
On second thought, this would suffer from the same thing I described below. Not deleting this endpoint object would leak the metric. Are you saying that because in this specific case, both endpoint have the same ID, that would result in the original endpoint object being deleted?

Reply:
Disregard my comments--was reading your comments from in order and didn't see your proposal at the end :+1: